instrumentation/weblogic/src/main/java/com/splunk/opentelemetry/middleware/weblogic/WebLogicAttributeCollector.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.middleware.weblogic;
+
+import com.splunk.opentelemetry.javaagent.bootstrap.MiddlewareHolder;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+
+public class WebLogicAttributeCollector {
+  private static final String CONTEXT_ATTRIBUTE_NAME = "otel.weblogic.attributes";
+  public static final String REQUEST_ATTRIBUTE_NAME = "otel.middleware";
+
+  public static void attachMiddlewareAttributes(HttpServletRequest servletRequest) {
+    WebLogicEntity.Request request = WebLogicEntity.Request.wrap(servletRequest);
+
+    Map<?, ?> attributes = fetchMiddlewareAttributes(request.getContext());
+    request.instance.setAttribute(REQUEST_ATTRIBUTE_NAME, attributes);
```

Comment:
Just wondering: have you considered adding a field to `MiddlewareHolder` and storing those attributes there instead of the request?

Reply:
The server and application fields may have different values in a single JVM, so the current static field approach `MiddlewareHolder` would not work there.